### PR TITLE
optimize mload instruction to not allocate a temporary vector

### PR DESCRIPTION
**Pull request by matklad**
Naked-eye based optimization -- it feels suboptimal to allocate a `Vec`
to fetch `[0; u32]`. Running uniswap bench in aurora shows that this is
an 1% improvement in wasm cost, so the compiler probably can't eliminate
this vec entirely.

### CODE REVIEW

**Comment by matklad** [comment]
@birchmd not sure what's the right workflow here to land the changes.

File: core/src/memory.rs [hunk 1]
```diff
@@ -96,6 +96,23 @@ impl Memory {
 		ret
 	}
 
+	/// Get `H256` from a specific offset in memory.
+	pub fn get_h265(&self, offset: usize) -> H256 {
```
Comment (olonho) [inline core/src/memory.rs:100]: `get_h256`

File: core/src/memory.rs [hunk 2]
```diff
@@ -96,6 +96,23 @@ impl Memory {
...
+		let mut ret = [0; 32];
+
+		#[allow(clippy::needless_range_loop)]
```
Comment (olonho) [inline core/src/memory.rs:103]: Maybe we could better do smth like
```
pub fn get_h256(&self, offset: usize) -> H256 {
		if offset + 32 < self.data.len() {
			H256(self.data.as_slice()[offset..offset + 32] as [u8;32])
		} else {
			H256::from_slice(&self.get(index, 32));
		}
	}
```

**Comment by birchmd** [comment]
> what's the right workflow here to land the changes.

This looks like a pretty simple change, should be easy to get it reviewed and merged if we make a PR against the upstream https://github.com/rust-blockchain/evm 
Once that is merged we can update our fork's master branch to be at parity with upstream again.

**Comment by matklad** [comment]
submitted upstream https://github.com/rust-blockchain/evm/pull/97

**Comment by birchmd** [comment]
@matklad can we close this since it is superseded by #7?

**Pull request closed without merging**